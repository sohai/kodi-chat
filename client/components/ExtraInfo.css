.root {
  height: 20px;
  padding: 5px;
}
