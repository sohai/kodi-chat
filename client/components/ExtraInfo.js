import React from 'react';
import {
  pure,
  compose,
  setDisplayName,
  setPropTypes,
  defaultProps
} from 'recompact';
import PropTypes from 'prop-types';
import Typography from './Typography';
import styles from './ExtraInfo.css';

export const ExtraInfoComponent = ({ info }) => (
  <Typography align="right" variant="caption" className={styles.root}>
    {info}
  </Typography>
);
export default compose(
  setDisplayName('ExtraInfoComponent'),
  setPropTypes({
    info: PropTypes.string
  }),
  defaultProps({
    info: ''
  }),
  pure
)(ExtraInfoComponent);
