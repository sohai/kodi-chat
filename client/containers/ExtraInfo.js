import React from 'react';
import { compose, pure, mapProps } from 'recompact';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import ExtraInfo from '../components/ExtraInfo';

const propTypes = PropTypes && {};

export const ExtraInfoRender = ({ info }) => <ExtraInfo info={info} />;

export const mapStateToProps = state => ({
  contact: state.users.contact.name,
  typing: state.users.contact.typing
});

ExtraInfoRender.propTypes = propTypes;
ExtraInfoRender.displayName = 'ExtraInfo';
export default compose(
  connect(mapStateToProps),
  mapProps(({ typing, contact }) => {
    return {
      info: typing ? `${contact} is typing` : ''
    };
  }),
  pure
)(ExtraInfoRender);
