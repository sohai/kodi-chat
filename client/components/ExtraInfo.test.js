/* global render, describe, it, expect */
/* eslint-disable import/first */
import React from 'react';
import ExtraInfo from './ExtraInfo';

describe('(Component) ExtraInfo', () => {
  it('should render', () => {
    const props = {
      info: 'Info text'
    };
    const wrapper = render(<ExtraInfo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
