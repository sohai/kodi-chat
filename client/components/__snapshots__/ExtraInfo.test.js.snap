// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ExtraInfo should render 1`] = `
<div
  class="root dark caption alignRight root"
>
  Info text
</div>
`;
